### PR TITLE
Fixing missing openApiNullable config for java apache-httpclient generator

### LINKED ISSUE #14827: [BUG] Java client apache-httpclient generator is missing the openApiNullable config/setting
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The Java Apache api-client has the ObjectMapper created without the `openApiNullable` config/module.

##### openapi-generator version

6.4.0

##### OpenAPI declaration file content or url

https://github.com/andrevegas/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/apache-httpclient/ApiClient.mustache#L133-L144

##### Steps to reproduce

use the mentioned api-client with Nullable request fields.

##### Suggest a fix

Similar to other java client generators https://github.com/andrevegas/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/native/ApiClient.mustache#L196-L198 the config should be part of the template.

### PR DESCRIPTION
**Pull request by andrevegas**
…
to close #14827

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: samples/client/echo_api/java/apache-httpclient/src/test/java/org/openapitools/client/CustomTest.java [hunk 1]
```diff
@@ -220,7 +220,7 @@ public void testDefaultValuesSerializationWithEmptyPayload() throws IOException
         Assert.assertNull(d.getArrayStringNullable());
         Assert.assertEquals(d.getArrayString().size(), 0);
 
-        Assert.assertEquals(apiClient.getObjectMapper().writeValueAsString(d), "{\"array_string_enum_ref_default\":[\"success\",\"failure\"],\"array_string_enum_default\":[\"success\",\"failure\"],\"array_string_default\":[\"failure\",\"skipped\"],\"array_integer_default\":[1,3],\"array_string\":[],\"array_string_nullable\":{\"present\":false},\"array_string_extension_nullable\":{\"present\":false},\"string_nullable\":{\"present\":false}}");
+        Assert.assertEquals(apiClient.getObjectMapper().writeValueAsString(d), "{\"array_string_enum_ref_default\":[\"success\",\"failure\"],\"array_string_enum_default\":[\"success\",\"failure\"],\"array_string_default\":[\"failure\",\"skipped\"],\"array_integer_default\":[1,3],\"array_string\":[]}");
```
Comment (andrevegas) [inline samples/client/echo_api/java/apache-httpclient/src/test/java/org/openapitools/client/CustomTest.java:223]: didn't want to change all the assertions.. but aren't those assertEquals params inverted?
expected should be the first param and the actual value right after/second param.

**Review by andrevegas — COMMENTED** [review]
@bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608

**Comment by andrevegas** [comment]
Hi @wing328, wondering if this PR/fix make sense and if we can get this in and release in the next minor release.

Thanks.

**Comment by wing328** [comment]
LGTM

File: samples/client/echo_api/java/apache-httpclient/src/test/java/org/openapitools/client/CustomTest.java (hunk 1 above)
Comment (wing328) [reply samples/client/echo_api/java/apache-httpclient/src/test/java/org/openapitools/client/CustomTest.java:223]: you're right: https://junit.org/junit4/javadoc/4.8/org/junit/Assert.html#assertEquals(java.lang.Object,%20java.lang.Object)

feel free to submit another PR to correct it. thanks for pointing it out.

**Pull request merged**